shopinvader_promotion_rule/models/sale_order.py

```diff
@@ -0,0 +1,18 @@
+# -*- coding: utf-8 -*-
+# Copyright 2020 ACSONE SA/NV (<http://acsone.eu>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import api, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    @api.multi
+    def reset_price_tax(self):
+        """
+        Inherit to apply the promotion rules when prices are updated
+        :return:
+        """
+        result = super(SaleOrder, self).reset_price_tax()
+        self.apply_promotions()
```

Comment:
@acsonefho IMO You must check if the SO has promotion rules before calling the apply_promotions.